src/Controllers/GagePageController.ts

```diff
@@ -884,6 +1104,772 @@ module StreamStats.Controllers {
                 }
                 });
             }
+            //checking for the latest year between the peaks and the daily flow
+            let finalPeakorDailyDate = new Date('January 1, 1800') // assign way in past
+            if (this.formattedPeakDates.length > 0) {
+                var finalPeakIndex = this.formattedPeakDates.length-1;
+                if (this.formattedPeakDates[finalPeakIndex].x > finalPeakorDailyDate) {
+                    finalPeakorDailyDate = this.formattedPeakDates[finalPeakIndex].x
+                }
+            }
+            if (this.formattedDailyFlow.length > 0) {
+                var finalDailyIndex = this.formattedDailyFlow.length-1;
+                if (this.formattedDailyFlow[finalDailyIndex].x > finalPeakorDailyDate) {
+                    finalPeakorDailyDate = this.formattedDailyFlow[finalDailyIndex].x
+                }
+            }
+            var finalIndex = this.formattedDailyFlow.length-1;
+            var finalYear = (finalPeakorDailyDate).getUTCFullYear(); //currently using the final daily flow point as latest year, but sometimes gage has more recent peaks
+            //return an array with a date each day between first recording and last
+            function dateRange(startDate, endDate, steps = 1) {
+                const dateArray = [];
+                let currentDate = new Date(startDate);
+                while (currentDate <= new Date(endDate)) {
+                    dateArray.push(new Date(currentDate));
```

Comment:
`currentDate` is already a Date object, so I think you can remove the `new Date()`. This also gets rid of the error on this line.

```suggestion
                    dateArray.push(currentDate);
```-